spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -154,7 +154,8 @@ public void close() {
 			if (this.readingRaw.get() && !finalizeRaw() && LOGGER.isWarnEnabled()) {
 				LOGGER.warn("Finalize on readRaw() returned false for " + this);
 			}
-			this.client.logout();
+			if (this.client.isConnected())
```

Comment:
Checkstyle violation:
```
[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java:157: 'if' construct must use '{}'s. [NeedBraces]
```

You can verify yourself locally using `gradlew clean :spring-integration-ftp:check`